mapie/conformity_scores/conformity_scores.py

```diff
@@ -0,0 +1,177 @@
+from abc import ABCMeta, abstractmethod
+
+import numpy as np
+
+from mapie._typing import ArrayLike, NDArray
+
+
+class ConformityScore(metaclass=ABCMeta):
+    """
+    Base class for conformity scores.
+
+    Warning: This class should not be used directly.
+    Use derived classes instead.
+
+    Parameters
+    ----------
+    sym: bool
+        Whether to consider the conformity score as symmetrical or not.
+
+    consistency_check: bool, optional
+        Whether to check the consistency between the following methods:
+        - ``get_estimation_distribution`` and
+        - ``get_signed_conformity_scores``
+
+        By default ``True``.
+
+    eps: float, optional
+        Threshold to consider when checking the consistency between the
+        following methods:
+        - ``get_estimation_distribution`` and
+        - ``get_signed_conformity_scores``
+        The following equality must be verified:
+        ``self.get_estimation_distribution(
+            y_pred, self.get_conformity_scores(y, y_pred)
+        ) == y``
+        It should be specified if ``consistency_check==True``.
+
+        By default ``np.float64(1e-8)``.
+    """
+
+    def __init__(
+        self,
+        sym: bool,
+        consistency_check: bool = True,
+        eps: np.float64 = np.float64(1e-8),
+    ):
+        self.sym = sym
+        self.consistency_check = consistency_check
+        self.eps = eps
+
+    @abstractmethod
+    def get_signed_conformity_scores(
+        self,
+        y: ArrayLike,
+        y_pred: ArrayLike,
+    ) -> NDArray:
+        """
+        Placeholder for ``get_signed_conformity_scores``.
+        Subclasses should implement this method!
+
+        Compute the signed conformity scores from the predicted values
+        and the observed ones.
+
+        Parameters
+        ----------
+        y: NDArray
```

Comment:
I think this might be recurrent, but in the parameters, you call this an `NDArray` but it's an `ArrayLike`. Also you are not give the shape of these arrays.